app/assets/javascripts/components/discussion_form.es6.jsx

```diff
@@ -177,6 +177,10 @@ class DiscussionForm extends React.Component {
   }
 
   renderContent() {
+    var editButton = null;
+    if (!this.props.current_user.is_admin || this.state.currentUser.id == this.state.data.user_id) {
```

Comment:
nit: delete currentUser from the state -- currentUser shouldn't be part of the state - just use it directly from the props passed in